### DISCUSSION TITLE
blender causes blurry final result image

### DISCUSSION THREAD

**Discussion by markputrella74** [OpenStitching/stitching#196] [Q&A] [answered]
Hi All !

First thanks to share to everyone this script !

I'm using with good results the command :
stitch C:\tmp\PhotoStich\3\1.jpg C:\tmp\PhotoStich\3\2.jpg --output C:\tmp\PhotoStich\1\result.jpg --confidence_threshold 0.2 --matches_graph_dot_file matches.txt  --detector sift --matches_graph_dot_file matches.txt --verbose --final_megapix -1
stitching C:\tmp\PhotoStich\3\1.jpg C:\tmp\PhotoStich\3\2.jpg into 20240330_141240_verbose_results

I'm testing stitching with slabs marble images.
I need to test the algo with the final goal to have just one pictures starting from two different photo shoot.
This will be an automatic system so , camera always in the same position, always the same distance, the slab just pass, and we'll take two shoot in two different slabs position.

Please find a we transfer link in which you will find all the verbose folder and orginal pictures for your reference.
I have noticed  that the final image is blurry respect the original one.
I mean is also blurry if compared with : 04_warped_img1, 08_compensated1 or 08_seam_mask1.jpg etc....
So in my opinion you can compare starting from this and you don't need to open the "original picures"
This happens in several images attached.

Very important !
This is seems NOT happening with the 20240330_143127_verbose_results folder !

Verbose folder:
https://we.tl/t-57zpT38qHm?utm_campaign=TRN_TDL_05&utm_source=sendgrid&utm_medium=email&trk=TRN_TDL_05

Orginal pictures
https://we.tl/t-ctmVD4oaxR?utm_campaign=TRN_TDL_05&utm_source=sendgrid&utm_medium=email&trk=TRN_TDL_05

For us is very import to preserve "quality" and details.
Many thanks for your help !
Matteo

**Comment by lukasalexanderweber** [comment]
Might be related to https://github.com/OpenStitching/stitching/discussions/84

**Reply by markputrella74** [reply]
Hi , thank you for the suggestion,
but looking to post you mentioned :
https://github.com/OpenStitching/stitching/discussions/84
seems there is no solution founded, am I wrong ?

In my case, i need to stitch always just two images.
And  looking to the images created during the process inside the verbose folder, seems that they have got a good quality, only the final one is affected.
Thank you very much again for your support.
Best regards,
Matteo

**Reply by markputrella74** [reply]
Furthermore to provide you more precise examples ,

if you look at the orginal and the result.jpg attached, you should easy find that the orginal one is better in terms of resolution and contrast.
The result.jpg seems quite blurry and lost some "details".

all the images inside the verbose folder are good (attached some example) 
:
![09_result_with_seam_lines](https://github.com/OpenStitching/stitching/assets/83245692/6d2b733c-3ebf-4b85-adfb-d477266f290a)
![605797-000025-QA-02CM_S1_c](https://github.com/OpenStitching/stitching/assets/83245692/194ec0ae-12ce-45cb-a3d1-7f0f64527178)
![08_seam_mask2](https://github.com/OpenStitching/stitching/assets/83245692/d6d003bc-3149-41a5-8144-09b2501a6ab6)
![08_compensated2](https://github.com/OpenStitching/stitching/assets/83245692/88f129e3-7509-4695-bca5-a433d2c4e75c)
![05_timelapse_img2](https://github.com/OpenStitching/stitching/assets/83245692/51b86b27-e57f-4eec-987c-523fe92f78cb)


Many thanks again for your support !!!!
Matteo

**Reply by lukasalexanderweber** [reply]
I've downloaded the files and can't follow you. The "Orginal pictures" are already stitched?

**Reply by markputrella74** [reply]
Hi,
thank you for your reply !

The original picture (the second from the top) is just the original photo taken from the camera, the pictures of the full slab, in one shot.
We have created the "left" and "right" pictures with Adobe Photoshop manually to simulate the behavior of the stitching algo.

Furthermore If you need more example to investigate you have got we transfer files at the top of this post as well.

Many thanks again for your support, this task is very important to us.
Best regards,
Matteo

**Reply by lukasalexanderweber** [reply]
But in "real" you'll be using 2 cameras or how will your setting look like?

**Reply by markputrella74** [reply]
no, in the final set up we'll use just one camera, and we'll take two pictures of the same slab while  passing on the roller. 

We are simulating now for a test.  
We'll use canon 6d full frame with sigma 24mm art, and we have got a python script that make lens calibration before "passing" the image to your algo. 

All the pictures we'll be taken at the same camera position with a steady tower.
Have a look here if interested to the project :
[https://iridefoto.it/wp/photo-scanner-slabs/](url)

Many thanks for your support.
very appreciated
Regards,
Matteo

**Reply by lukasalexanderweber** [reply]
> better in terms of resolution and contrast

the resolution is the same, there are some pixel offsets but I guess they come from cropping. For the contrast, you might want to switch off [Exposure Compensation](https://docs.opencv.org/3.4/dd/dff/group__stitching__exposure.html), since this approach changes pixel values in your whole image. You don't want to find matches at the "rolls" of your assembly line. You could try to detect the plates and only use these to find features (use Feature Masking). Of course the plates need to have enough features to be stitched. 

There was a similar approach [here](https://github.com/OpenStitching/stitching/discussions/81)

**Reply by markputrella74** [reply]
Hi Lukas !
Thank you again for your reply !!

Resolution in terms of pixels are of course the same but final img is less sharply and blurry .
We have tried again with the option : " --compensator no" , with the same result.
I have attached an img to help to understand what happens in the final img and compare.
As my previous message looking in the verbose folder it seems that all the sequences from 01 to 08 have preserved the original "quality" and only the 09_result is affected.

We don't find any documentation to help implement the "Feature Masking  " could you please give us any details ?
Many thanks again for your support !Regards,Matteo
![09_result copia](https://github.com/OpenStitching/stitching/assets/83245692/13a8a3be-f3e7-410c-b7e4-a552feab64e1)
![particolareOrig](https://github.com/OpenStitching/stitching/assets/83245692/88eda710-b7de-46c0-aa5e-9def0e93daa4)
![particolare](https://github.com/OpenStitching/stitching/assets/83245692/bce7f022-dbe5-4f16-9e01-468efb760d95)

**Comment by lukasalexanderweber** [comment]
At the end of the tutorial is a section about masking. Maybe try another image format (PNG, tiff), as jpg might compress Details

**Reply by markputrella74** [reply]
Hi Lukas,
Is there a possibility to contact you directly?
We will need to modify this script and we will be able to invest in this .

We didn't find your pubblic email so we can leave our contact :

Matteo : support@iridefoto.it
www.iridefoto.it

Many thanks again.
Matteo

**Reply by markputrella74** [reply, marked as answer]
Hi,
don't ask me why but problem seems resolved with the following option :

--blender_type no 

We have tryed all the images and now quality is ok.
we use at the end :

stitch C:\tmp\PhotoStich\1\1.jpg C:\tmp\PhotoStich\1\2.jpg --output C:\tmp\PhotoStich\1\result.jpg --confidence_threshold 0.2  --detector sift   --blender_type no

**Reply by lukasalexanderweber** [reply]
Great, good luck with your project!
